src/rpc/rawtransaction_util.cpp

```diff
@@ -283,8 +289,16 @@ UniValue SignTransaction(interfaces::Chain& chain, CMutableTransaction& mtx, con
     bool fComplete = vErrors.empty();
 
     UniValue result(UniValue::VOBJ);
-    result.pushKV("hex", EncodeHexTx(CTransaction(mtx)));
+    std::string hex = EncodeHexTx(CTransaction(mtx));
+    result.pushKV("hex", hex);
     result.pushKV("complete", fComplete);
+    if (known_inputs) {
+        for (const CTxOut& txout : mtx.vout) {
+            inout_amount -= txout.nValue;
+        }
+        result.pushKV("fee", ValueFromAmount(inout_amount));
+        result.pushKV("feerate", ValueFromAmount(inout_amount*1000/(hex.length()>>1)));
```

Comment:
should use the virtual size, not the tx size

Reply:
@NicolasDorier I changed to use vsize:
```
{
  "hex": "0200000000010104f1e2bcda677ec9c76c4201ce5ab1dec28b2127fa5cc1a0555401887ed1a49b00000000171600143f7d5c060647cf38a896d2b0ad77004526eead1bfeffffff02109242060100000017a914b7036cfe5ee607f473630a51911f60f99c85260f870065cd1d0000000017a914bd8a3d83ca6e3cfa74386c0e61374a62e7eba4178702473044022058bf8809ebd3816a8270d0bd740a26f962d8deab02a7a46c008b4de7525bb15c022005dd8381cd9cd1eb48be647ccdb098d7e57a6d1fdc354911348427174e4ed43a012102cd5dd2f7d61b58a8f0a319dbd89a35be73937003e6e88313cb04e7b4bfd56fc400000000",
  "complete": true,
  "fee": 0.00003320,
  "feerate": 0.00020000
}
```
with
```
  "vsize": 166,
```
gives
```R
> vsize=166
> fee=0.00003320
> 1000*(fee/vsize)
[1] 2e-04
```
(`2e-04` = `0.00020000`)